src/scripts/core/models/button.ts

```diff
@@ -12,10 +12,10 @@ export class Button extends Container {
         this.interactive = true;
         this.cursor = "pointer";
 
-        this.on('pointerdown',this.onPointerDown, this)
-        .on('pointerover', this.onPointerOver, this)
-        .on('pointerup', this.onPointerUp, this)
-        .on('pointerout', this.onPointerOut, this);
+        this.on('pointerdown',this.buttonActivateAndCallback, this)
+        .on('pointerover', this.highlightAndEnlargeButton, this)
+        .on('pointerup', this.resetButtonToStartingState, this)
+        .on('pointerout', this.resetButtonToStartingState, this);
```

Comment:
I've refactored the code as per earlier PR discussion

onPointerUp and onPointerOut were two functions running the same code so I've changed the function names to streamline the calls.